File: nicolaspuebla_proyecto_final/src/main/java/com/example/nicolaspuebla_proyecto_final/model/dto/TokenCheckResponse.java
package com.example.nicolaspuebla_proyecto_final.model.dto;

import com.example.nicolaspuebla_proyecto_final.model.dataModels.MobileUser;

/**
 * Clase que representa la respuesta para la verificación de un token de usuario móvil.
 */
public class TokenCheckResponse {
    private MobileUser user;

    public TokenCheckResponse(){}

    public TokenCheckResponse(MobileUser user) {
        this.user = user;
    }

    public MobileUser getUser() {
        return user;
    }

    public void setUser(MobileUser user) {
        this.user = user;
    }    
}

File: nicolaspuebla_proyecto_final_android/app/src/main/java/com/example/nicolaspuebla_proyecto_final_android/data/model/auth/TokenCheckResponse.kt
package com.example.nicolaspuebla_proyecto_final_android.data.model.auth

import com.example.nicolaspuebla_proyecto_final_android.data.model.dataClases.MobileUser

/**
 * Clase empleada para recibir la respuesta del microsercivio tras comprobar la valided de un token.
 */
data class TokenCheckResponse (
    val user: MobileUser?
)